packages/labs/preact-signals/src/lib/signal-watcher.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type {ReactiveElement} from 'lit';
+import {effect} from '@preact/signals-core';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ReactiveElementConstructor = new (...args: any[]) => ReactiveElement;
+
+/**
+ * Adds the ability for a LitElement or other ReactiveElement class to
+ * watch for access to Preact signals during the update lifecycle and
+ * trigger a new update when signals values change.
+ */
+export function SignalWatcher<T extends ReactiveElementConstructor>(
+  Base: T
+): T {
+  return class SignalWatcher extends Base {
+    private __dispose?: () => void;
+
+    override performUpdate() {
+      // ReactiveElement.performUpdate() also does this check, so we want to
+      // also bail early so we don't erroneously appear to not depend on any
+      // signals.
+      if (!this.isUpdatePending) {
+        return;
+      }
+      // If we have a previous effect, dispose it
+      this.__dispose?.();
```

Comment:
Should we dispose on disconnect as well?

Reply:
done